From: Alessandro Astone <alessandro.astone@canonical.com>
Date: Thu, 12 Feb 2026 12:56:09 +0100
Subject: vulkan: Fix type mismatch on 32-bit

timestamp_pool_nodes is declared as gsize* but allocated as uint64_t*, which
expand to different types on 32-bit architectures and emit a compiler warning.

Forwarded: https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/9460
---
 gsk/gpu/gskvulkandebugframe.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gsk/gpu/gskvulkandebugframe.c b/gsk/gpu/gskvulkandebugframe.c
index 160cd68..e27bfa8 100644
--- a/gsk/gpu/gskvulkandebugframe.c
+++ b/gsk/gpu/gskvulkandebugframe.c
@@ -80,7 +80,7 @@ gsk_vulkan_debug_frame_submit_ops (GskVulkanFrame        *frame,
       self->pool_size = 3 * self->n_ops / 2;
 
       self->timestamp_pool_values = g_new (uint64_t, self->pool_size * 2);
-      self->timestamp_pool_nodes = g_new (uint64_t, self->pool_size);
+      self->timestamp_pool_nodes = g_new (gsize, self->pool_size);
       GSK_VK_CHECK (vkCreateQueryPool, vk_device,
                                        &(VkQueryPoolCreateInfo) {
                                            .sType = VK_STRUCTURE_TYPE_QUERY_POOL_CREATE_INFO,
